task/config.go

```diff
@@ -0,0 +1,65 @@
+package task
```

Comment:
Putting this in `task/config.go`  sounds like this is a task configuration but it is rather a repository configuration right? Should the file be renamed from `config.go` to `repository.go`? Or even put this in a `repository` package e.g. `repository/config.go`?

Reply:
The interface for `Repository` is also defined within `task`. That's why I moved it there. So we would probably want to move that one as well. Would leave it for now.

Reply:
Let's leave it for now then